Submitted By: Jeremy Huntwork (jhuntwork at linuxfromscratch dot org)
Date: 2005-09-18
Initial Package Version: 2.3.5
Origin: lists.debian.org
Upstream Status: Applied
Description: Fixes compiling of glibc-2.3.5 with gcc-4 on powerpc machines.

diff -Naur glibc-2.3.5.orig/sysdeps/unix/sysv/linux/powerpc/sys/procfs.h glibc-2.3.5/sysdeps/unix/sysv/linux/powerpc/sys/procfs.h
--- glibc-2.3.5.orig/sysdeps/unix/sysv/linux/powerpc/sys/procfs.h	2003-12-17 18:09:34.000000000 -0500
+++ glibc-2.3.5/sysdeps/unix/sysv/linux/powerpc/sys/procfs.h	2005-09-18 09:13:01.458306384 -0400
@@ -46,15 +46,10 @@
 typedef double elf_fpreg_t;
 typedef elf_fpreg_t elf_fpregset_t[ELF_NFPREG];
 
-/* gcc 3.1 and newer support __uint128_t.  */
-#if !__GNUC_PREREQ(3,1)
-typedef struct {
-  unsigned long u[4];
-} __attribute((aligned(16))) __uint128_t;
-#endif
-
 /* Altivec registers */
-typedef __uint128_t elf_vrreg_t;
+typedef struct {
+  unsigned int u[4];
+} __attribute__ ((aligned (16))) elf_vrreg_t;
 typedef elf_vrreg_t elf_vrregset_t[ELF_NVRREG];
 #endif
 

